02-peer-protocol.md

```diff
@@ -1127,6 +1190,17 @@ A receiving node:
   current commitment transaction:
     - SHOULD fail the channel,
       - but MAY delay this check until the `update_fee` is committed.
+  - if the `dust_balance_on_counterparty_tx` at the new `dust_buffer_feerate` is superior to `max_dust_htlc_exposure_msat`:
+    - MAY fail the channel
```

Comment:
```suggestion
    - SHOULD fail the channel
```

Reply:
See the rational just under. That's a MAY as automatic-closure might bring newer issues. A third-party to a channel could relay and withhold HTLCs across the link to provoke a close.

See also Eclair new config settings `close-on-update-fee-dust-exposure-overflow` : https://github.com/ACINQ/eclair/pull/1985/files, which is false by default. Though in LDK we do auto-close by default.

As it's arbitrating between two safety risks, I think it's wiser to defer the choice to the implementator.

Reply:
Well 'SHOULD' is still a deferral to the implementor, just a more strongly worded form!